r1cs/gadgets/crypto/src/prf/ripemd160.rs

```diff
@@ -0,0 +1,458 @@
+//! Circuits for the [RIPEMD160] hash function
+//!
+//! [RIPEMD160]: https://tools.ietf.org/html/rfc6234
+
+use r1cs_std::boolean::Boolean;
+use r1cs_std::eq::MultiEq;
+use r1cs_std::uint32::UInt32;
+use r1cs_std::uint8::UInt8;
+use r1cs_core::{ConstraintSystem, SynthesisError};
+use algebra::PrimeField;
+
+use crate::sha256::{sha256_ch_boolean, triop};
+
+/// Outputs K[round_idx] and K'[round_idx]
+fn get_round_constants(round_idx: usize) -> (UInt32, UInt32) {
+    let (k, k_prime): (u32, u32) = match round_idx {
+        round_idx if round_idx <= 15 =>                    (0x00000000, 0x50a28be6),
+        round_idx if round_idx >= 16 && round_idx <= 31 => (0x5a827999, 0x5c4dd124),
+        round_idx if round_idx >= 32 && round_idx <= 47 => (0x6ed9eba1, 0x6d703ef3),
+        round_idx if round_idx >= 48 && round_idx <= 63 => (0x8f1bbcdc, 0x7a6d76e9),
+        round_idx if round_idx >= 64 && round_idx <= 79 => (0xa953fd4e, 0x00000000),
+        _ => unreachable!()
+    };
+    (UInt32::constant(k), UInt32::constant(k_prime))
+}
+
+#[allow(clippy::unreadable_literal)]
+const IV: [u32; 5] = [
```

Comment:
Comment: these are the initial values of h0, h1, h2, h3, h4 in the description of RIPEMD160